src/DIRAC/Core/Base/Client.py

```diff
@@ -147,14 +147,11 @@ def addFunctions(clientCls):
         """Add the functions to the decorated class."""
         attrDict = dict(clientCls.__dict__)
         for extension in extensionsByPriority():
+            moduleBase = f"{extension}.{systemName}System.Service"
+            fullHandlerClassPath = f"{extension}.{handlerClassPath}"
             try:
-                path = importlib_resources.path(
-                    "%s.%sSystem.Service" % (extension, systemName),
-                    "%s.py" % handlerModuleName,
-                )
-                fullHandlerClassPath = "%s.%s" % (extension, handlerClassPath)
-                with path as fp:
-                    handlerAst = ast.parse(fp.read_text(), str(path))
+                path = importlib_resources.files(moduleBase) / f"{handlerModuleName}.py"
+                handlerAst = ast.parse(path.read_text(), f"{moduleBase}.{handlerModuleName}")
```

Comment:
Can you compare what's done here wrt what is done in https://github.com/DIRACGrid/DIRAC/pull/6062/files#diff-1cee8c07333a9b6b8ad264693ea3184e5670fce1038a451fd8190748d864f26f ?

Reply:
The two diffs are unrelated and that one is just changing `importlib_resources` to the standard library implementation (`importlib.resources`).